appserver/payara-appserver-modules/microprofile/config/src/main/java/fish/payara/microprofile/config/cdi/ConfigPropertyProducer.java

```diff
@@ -111,4 +112,14 @@ public static final Object getGenericProperty(InjectionPoint ip) {
         return result;
     }
 
+    private static Class<?> getElementTypeFrom(ParameterizedType collectionType) {
+        Type elementType = collectionType.getActualTypeArguments()[0];
+        if (!(elementType instanceof Class)) {
+            throw new DeploymentException(
```

Comment:
I was looking into not using `getSetValues` and `getListValues` directly (plain API access) as I realised that these should be implemented by using array access and later wrapping the returned array in a `List` or `Set`. What is unclear is how the default gets past on using the standard API. So I made this cleanup as a preparation and noticed that it was simply assumed that the actual type argument is a `Class` type which it might not be so I added a more descriptive exception instead of the `ClassCastException` that was caused in such a case. As far as I understand it using a non `Class` type is not allowed according to standard but a user might get the idea it is valid and use a type like `List<List<String>>`.